lib/icingadb/icingadb-objects.cpp

```diff
@@ -242,12 +245,20 @@ void IcingaDB::UpdateAllConfigObjects()
 				if (lcType != GetLowerCaseTypeNameDB(object))
 					continue;
 
-				CreateConfigUpdate(object, lcType, hMSets, publishes, false);
+				std::vector<Dictionary::Ptr> runtimeUpdates;
+				CreateConfigUpdate(object, lcType, hMSets, runtimeUpdates, false);
 
 				// Write out inital state for checkables
 				if (dumpState) {
-					states.emplace_back(GetObjectIdentifier(object));
-					states.emplace_back(JsonEncode(SerializeState(dynamic_pointer_cast<Checkable>(object))));
+					String objectKey = GetObjectIdentifier(object);
+					Dictionary::Ptr state = SerializeState(dynamic_pointer_cast<Checkable>(object));
+					String checksum = HashValue(state);
```

Comment:
So in the end, this hash will decide when Icinga DB will update `host_state`/`service_state` in MySQL, right?

Reply:
Correct, it's used during the initial sync got check for changes.